### ISSUE TITLE
Variables to enhance notifications

### ISSUE THREAD

**Issue by JChipault** [USGS-WiM/whispersservices#548]
**Enhancement**
After some user feedback, we find we'd like some more dynamic variables made available in the notifications. Neil can change the static text, so it's just making these extra variables available that we're looking for. Holler if any of these are complicated. Thanks!

ID 1: 
High Impact Diseases
Variables to add: 
First name, last name, user name, and organization for person who entered the high impact disease
First name, last name, user name, and organization for the event owner
Species to which the high impact diagnosis was attached

ID 2: 
User Change Request
Variables to add:
Email of the requester

ID 3:
User Change Request Response Yes
Variables to add:
First name, last name, and user name of the requester

ID 6:
Collaborator Added
Variables to add:
Organization of the person who added the collaborator

ID 9:
Service Request Response
Variables to add:
First name, last name, user name, and organization for the person who responded
First name, last name, user name, and organization for the person who initiated the service request

ID 10:
Service Request Comment
Variables to add:
First name, last name, user name, and organization for the person who commented
First name, last name, user name, and organization for the person who initiated the service request

ID 12:
Your Events
Variables to add:
user name and organization for the person who updated the record

ID 13:
Organization Events
Variables to add:
user name for the person who updated the record

ID 14:
Collaborator Events
Variables to add:
user name and organization for the person who updated the record

ID unknown:
Also, I think this text might be hard-coded, but would also like to add Event ID to the Third Party Service Validation Warning notifications that currently tell me when a lat/long isn't in the county or country expected. Is this possible?

**Comment by aaronstephenson** [comment]
@JChipault I can quickly add these to the code, but:

1. I need to know if the new variables go in the body template or subject template or both, and the best way to know this would be for you to just supply me with your new body and subject templates.
2. The code changes need to be simultaneously deployed with changes to the "whispers_notificationmessagetemplate" table's "message_variable", "body_template", and "subject_template" columns, otherwise the app will break if either the code changes or the DB table changes are deployed before the other one.

**Comment by aaronstephenson** [comment]
I added Event ID to the Third Party Service Validation Warning email bodies. That change will be deployed along with the eventual notification message variable changes.

**Comment by JChipault** [comment]
Regarding 1 above - Just the body. We haven't made the new language yet (waiting to see if adding these variables was do-able) so don't have that to pass to you, but I don't believe we have any changes in mind to include these variables for the subject line. 

Regarding 2 above - @nbaertlein I'm guessing this pertains to the changes you've recently made to text on the test site??? Maybe? Or maybe I'm off.

**Comment by aaronstephenson** [comment]
Everything is ready to go on my end, got PR #549 ready to merge. Just need to coordinate the "whispers_notificationmessagetemplate" table changes with @nbaertlein so the app doesn't break.

**Comment by JChipault** [comment]
I don't quite follow... are you saying you didn't do ID 13 because that one is a big lift?

**Comment by JChipault** [comment]
Got it. Thanks for explaining more. 

So the updated text that Neil was working on loading doesn't have these new variables. Do we need to update the text again before your change are deployed so that the new variables on this ticket are integrated in to the text now?

**Comment by aaronstephenson** [comment]
The two things (deploying the new code and updating the template text in the DB table) must be done simultaneously, or there will be errors caused by the mismatch of variables and no notifications will be sent. If you could supply me with the new template text, I could do both things together.

**Comment by JChipault** [comment]
Ok, thanks. We'll work on updating text next week. I didn't realize both would have to be deployed simultaneously so thanks for pointing that out.

**Comment by nbaertlein** [comment]
@aaronstephenson I do not think adding the message variables to the code and table, but not the message body and subject templates will cause errors.  Case in point:  I just tested the collaboration request (template id=5) with all the variables and without all variables and still got the email.  The message variables for this are {first_name,last_name,organization,event_id,comment,email}
Successful message with all variables:
![image](https://user-images.githubusercontent.com/38669574/175997381-6938ae0c-85b1-452b-bb67-b950fd76fa13.png)

Successful message with no {last_name} or {email}:
![image](https://user-images.githubusercontent.com/38669574/175997471-65680407-7e4f-454b-bb90-cc801ddb76f7.png)

The main reason I wanted to investigate is that we recently updated all the message templates in the test site.  There are quite a few changes and not all of them work yet.  I'd like to get these to work first, then introduce the new variables.  Otherwise I feel there may be too many moving pieces when troubleshooting.

Does it work for you to hold on pushing the new variables until we get our current batch of message updates working?
